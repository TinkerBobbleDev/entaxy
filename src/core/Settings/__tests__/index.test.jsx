import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { SettingsComponent } from '..'

jest.mock('../form', () => 'SettingsForm')
const mochSaveSettings = jest.fn()
const mochDeleteAllData = jest.fn()
const mochShowSnackbar = jest.fn()

describe('Settings', () => {
  it('matches snapshot with no accounts and no selected accountId', () => {
    const component = renderer.create((
      <SettingsComponent
        history={{}}
        classes={{}}
        deleteAllData={mochDeleteAllData}
        saveSettings={mochSaveSettings}
        showSnackbarMessage={mochShowSnackbar}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Component methods', () => {
    const mochHistoryPush = jest.fn()
    const wrapper = shallow((
      <SettingsComponent
        history={{ push: mochHistoryPush }}
        classes={{}}
        deleteAllData={mochDeleteAllData}
        saveSettings={mochSaveSettings}
        showSnackbarMessage={mochShowSnackbar}
      />
    ))
    const instance = wrapper.instance()

    it('saves settings and shows the snackbar', async () => {
      await instance.handleSave()
      expect(mochSaveSettings).toHaveBeenCalled()
      expect(mochShowSnackbar).toHaveBeenCalledWith({ status: 'success', text: 'Your settings have been saved' })
    })

    it('deletes all data', async () => {
      await instance.handleResetData()
      expect(mochDeleteAllData).toHaveBeenCalled()
      expect(mochHistoryPush).toHaveBeenCalledWith('/dashboard')
    })

    it('cancels and returns to dashboard', async () => {
      instance.handleCancel()
      expect(mochHistoryPush).toHaveBeenCalledWith('/dashboard')
    })
  })
})
