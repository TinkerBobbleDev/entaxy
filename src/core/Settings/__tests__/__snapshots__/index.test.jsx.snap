// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot with no accounts and no selected accountId 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
      >
        Settings
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <SettingsForm
      handleDeleteAllData={[Function]}
      handleSave={[Function]}
    />
  </div>
</div>
`;
